dev-python/pytest-asyncio/pytest-asyncio-0.9.0.ebuild

```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{5,6} )
+
+inherit distutils-r1
+
+DESCRIPTION="Pytest support for asyncio"
+HOMEPAGE="https://github.com/pytest-dev/pytest-asyncio https://pypi.org/project/pytest-asyncio/"
+SRC_URI="https://github.com/pytest-dev/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="
+	dev-python/async_generator[${PYTHON_USEDEP}]
```

Comment:
Hi! I feel I am in some kind of a deadend. If you are not willing to help me out, it's ok, I will see my ways out of it, but in case there's something absolutely stupid/wrong I am doing, maybe your 5 mins can save my day of work.

So while I was working on introducing support for Python3.7 in dev-python/async_generator, I discovered `async_generator-1.10.ebuild` wouldn't pass tests on my local gentoo and I have tried testing an untouched ebuild. I get the following output
```
>>> Source compiled.
>>> Test phase: dev-python/async_generator-1.10
 * python3_5: running distutils-r1_run_phase python_test
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.5.5, pytest-3.8.2, py-1.5.4, pluggy-0.7.1 -- /usr/bin/python3.5
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10/.hypothesis/examples')
rootdir: /var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10, inifile:
plugins: forked-0.2, asyncio-0.9.0, hypothesis-3.74.3
collecting ... collected 0 items / 1 errors

===================================================================================================== ERRORS ======================================================================================================
________________________________________________________________________________________________ ERROR collecting  ________________________________________________________________________________________________
/usr/lib64/python3.5/site-packages/_pytest/config/__init__.py:382: in _getconftestmodules
    return self._path2confmods[path]
E   KeyError: local('/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10/async_generator/_tests')

During handling of the above exception, another exception occurred:
/usr/lib64/python3.5/site-packages/_pytest/config/__init__.py:413: in _importconftest
    return self._conftestpath2mod[conftestpath]
E   KeyError: local('/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10/async_generator/_tests/conftest.py')

During handling of the above exception, another exception occurred:
/usr/lib64/python3.5/site-packages/_pytest/config/__init__.py:419: in _importconftest
    mod = conftestpath.pyimport()
/usr/lib64/python3.5/site-packages/py/_path/local.py:686: in pyimport
    raise self.ImportMismatchError(modname, modfile, self)
E   py._path.local.LocalPath.ImportMismatchError: ('async_generator._tests.conftest', '/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10-python3_5/lib/async_generator/_tests/conftest.py', local('/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10/async_generator/_tests/conftest.py'))

During handling of the above exception, another exception occurred:
/usr/lib64/python3.5/site-packages/py/_path/common.py:377: in visit
    for x in Visitor(fil, rec, ignore, bf, sort).gen(self):
/usr/lib64/python3.5/site-packages/py/_path/common.py:429: in gen
    for p in self.gen(subdir):
/usr/lib64/python3.5/site-packages/py/_path/common.py:418: in gen
    dirs = self.optsort([p for p in entries
/usr/lib64/python3.5/site-packages/py/_path/common.py:419: in <listcomp>
    if p.check(dir=1) and (rec is None or rec(p))])
/usr/lib64/python3.5/site-packages/_pytest/main.py:562: in _recurse
    ihook = self.gethookproxy(path)
/usr/lib64/python3.5/site-packages/_pytest/main.py:416: in gethookproxy
    my_conftestmodules = pm._getconftestmodules(fspath)
/usr/lib64/python3.5/site-packages/_pytest/config/__init__.py:396: in _getconftestmodules
    mod = self._importconftest(conftestpath)
/usr/lib64/python3.5/site-packages/_pytest/config/__init__.py:436: in _importconftest
    raise ConftestImportFailure(conftestpath, sys.exc_info())
E   _pytest.config.ConftestImportFailure: ImportMismatchError('async_generator._tests.conftest', '/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10-python3_5/lib/async_generator/_tests/conftest.py', local('/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10/async_generator/_tests/conftest.py'))
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================= 1 error in 0.23 seconds =============================================================================================
 * ERROR: dev-python/async_generator-1.10::bucefal91-fork failed (test phase):
 *   tests failed under python3.5
 * 
 * Call stack:
 *     ebuild.sh, line  124:  Called src_test
 *   environment, line 2642:  Called distutils-r1_src_test
 *   environment, line  893:  Called _distutils-r1_run_foreach_impl 'python_test'
 *   environment, line  344:  Called python_foreach_impl 'distutils-r1_run_phase' 'python_test'
 *   environment, line 2175:  Called multibuild_foreach_variant '_python_multibuild_wrapper' 'distutils-r1_run_phase' 'python_test'
 *   environment, line 1547:  Called _multibuild_run '_python_multibuild_wrapper' 'distutils-r1_run_phase' 'python_test'
 *   environment, line 1545:  Called _python_multibuild_wrapper 'distutils-r1_run_phase' 'python_test'
 *   environment, line  556:  Called distutils-r1_run_phase 'python_test'
 *   environment, line  824:  Called python_test
 *   environment, line 2544:  Called die
 * The specific snippet of code:
 *       py.test -v || die "tests failed under ${EPYTHON}"
 * 
 * If you need support, post the output of `emerge --info '=dev-python/async_generator-1.10::bucefal91-fork'`,
 * the complete build log and the output of `emerge -pqv '=dev-python/async_generator-1.10::bucefal91-fork'`.
 * The complete build log is located at '/var/tmp/portage/dev-python/async_generator-1.10/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/dev-python/async_generator-1.10/temp/environment'.
 * Working directory: '/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10'
 * S: '/var/tmp/portage/dev-python/async_generator-1.10/work/async_generator-1.10'
```

I've spent one hour googling for the `KeyError` exception and haven't found any meaningful explanation to it. My local pytest does pass testing of other packages.

Reply:
@sbraz is an expert on weird pytest issues.

Reply:
@mgorny @sbraz Luckily, after sleeping a night on it I managed to solve it.

I am not sure why/how, but I managed to mess something up in local python environment. All the findings on the web about this kind of error messages led to people manually removing `*.pyc` files. So I spent some time to debug what *.pyc leftovers are not being removed from my `/usr/lib64/python*/` folders when I unmerge all related python packages. It did reveal a handful of pyc files. I removed them manually and it got back to work.

I just pushed a commit that revision bumps `dev-python/async_generator` and adds Python3.7 support into it. I confirm locally python3_7 passes tests in that package. Then I amended my original `dev-python/pytest-asyncio` commit to include Python3_7 useflag too.

Would you review once again my work to see if it's good for commit or I shall adjust something more? :)

Reply:
I can try to run the tests if that helps.

Reply:
Sure, eventually I believe i was able to pin-point a cause and it was local; but it would be nice to hear how they behave in another host.